projects/packages/sync/src/class-sender.php

```diff
@@ -418,12 +418,12 @@ public function do_dedicated_sync_and_exit( $do_real_exit = true ) {
 			session_write_close();
 		}
 
-		// Output not used right now. Try to release dedicated sync lock
-		Dedicated_Sender::try_release_lock_spawn_request();
-
 		// Actually try to send Sync events.
 		$result = $this->do_sync_and_set_delays( $this->sync_queue );
 
+		// Output not used right now. Try to release dedicated sync lock
+		Dedicated_Sender::try_release_lock_spawn_request();
```

Comment:
Reviewed the current code/P2s to see if I could understand why we would release before do_sync_and_set_delays but can't find anything. Makes sense this would be called after the sync is performed, especially as there is the possibility to spawn_sync immediately after.

Reply:
Sync has always been the issue where the edge cases tend to high performance/traffic/change sites and those are where we really need to optimize. Taking off pressure on host servers (php workers) is exactly what sync needs to do. Thanks for finding this.